packages/mds-transactions-service/repository/index.ts

```diff
@@ -41,10 +41,20 @@ class TransactionReadWriteRepository extends ReadWriteRepository {
     }
   }
 
-  // TODO search criteria, paging
-  public getTransactions = async (search: TransactionSearchParams): Promise<TransactionDomainModel[]> => {
+  // TODO search criteria
+  public getTransactions = async (
+    search: TransactionSearchParams
+  ): Promise<{ transactions: TransactionDomainModel[]; cursor: Cursor }> => {
     const { connect } = this
-    const { provider_id, start_timestamp, end_timestamp } = search
+    const { provider_id, start_timestamp, end_timestamp, before, after } = search
+    let { limit } = search
+    limit = limit || 1
+    if (Number.isSafeInteger(limit)) {
+      limit = Math.max(1, limit)
+      limit = Math.min(1000, limit)
+    } else {
+      limit = 10
+    }
```

Comment:
I would recommend just using Joi as a validator and doing this by:
```typescript
Joi.number().integer().min(1).max(1000).default(10)
```
(pretty sure that should work)

But, if you really want to do it manually, you should probably do something along the lines of:
```typescript
const getLimit = (search: TransactionSearchParams) => {
  const { limit = 10 } = search

  if (!Number.isSafeInteger(limit) || limit < 1 || limit > 1000) return 10

  return limit
}

const limit = getLimit()
```

If JS complains about `search` being shadowed or something, then you'll wanna put that validator/getter into a separate out-of-this-function-scope `validators` file